pkg/rc/replication_controller.go

```diff
@@ -228,12 +244,28 @@ func (rc *replicationController) meetDesires() error {
 	nodesChanged := false
 	switch {
 	case rc.ReplicasDesired > len(current):
+		if rc.nodeTransfer.quit != nil {
+			// Remove nodeTransfer.newNode from eligible during a node transfer
+			// because if it is scheduled again below in addPods(), the rc
+			// will have one less node than desired. Waiting for the next loop
+			// for this case again will add a node, but is a bad experience.
+			for i, v := range eligible {
+				if v == rc.nodeTransfer.newNode {
+					eligible = append(eligible[:i], eligible[i+1:]...)
+					break
+				}
+			}
+		}
 		err := rc.addPods(current, eligible)
 		if err != nil {
 			return err
 		}
 		nodesChanged = true
 	case len(current) > rc.ReplicasDesired:
+		if rc.nodeTransfer.quit != nil {
```

Comment:
I still think you need to do something like

```
rc.nodeTransfer.mu.Lock()
if rc.nodeTransfer.quit != nil {
    close(rc.nodeTransfer.quit)
}
rc.nodeTransfer.mu.Unlock()
```

and then below acquire the lock before doing `rc.nodeTransfer = nodeTransfer{}` which nils out the channel. Otherwise you might panic closing a nil channel

Reply:
Do you think I need a new mutex specifically for nodeTransfers or can I just use the `rc.mu` that the rc struct already has

Reply:
you could use the same one, yeah. They're protecting different data but there's not a lot of contention on that one so it's fine